x-pack/plugins/gis/public/shared/layers/sources/ems_tms_source.js

```diff
@@ -21,6 +26,31 @@ export class EMSTMSSource extends TMSSource {
     };
   }
 
+  static renderEditor({ dataSourcesMeta, onPreviewSource }) {
+
+    const emsTmsOptionsRaw = (dataSourcesMeta) ? dataSourcesMeta.ems.tms : [];
+    const emsVectorOptions = emsTmsOptionsRaw ? emsTmsOptionsRaw.map((file) => ({
```

Comment:
Then ternary operator is not needed. `emsTmsOptionsRaw` will always be an array and you can just call map on it.

Also, why is it called `emsTmsOptionsRaw` and then the map function results are called `emsVectorOptions`. How did the map function turn Tms into Vector? Maybe rename `emsVectorOptions` -> emsTmsOptions